samples/apps/copilot-chat-app/webapi/Auth/ApiKeyAuthenticationHandler.cs

```diff
@@ -16,41 +16,53 @@ public class ApiKeyAuthenticationHandler : AuthenticationHandler<ApiKeyAuthentic
     public const string AuthenticationScheme = "ApiKey";
     public const string ApiKeyHeaderName = "x-api-key";
 
-    // TODO: not used?
-    private readonly IOptionsMonitor<ApiKeyAuthenticationSchemeOptions> _options;
-
     /// <summary>
     /// Constructor
     /// </summary>
     public ApiKeyAuthenticationHandler(
         IOptionsMonitor<ApiKeyAuthenticationSchemeOptions> options,
-        ILoggerFactory logger,
+        ILoggerFactory loggerFactory,
         UrlEncoder encoder,
-        ISystemClock clock) : base(options, logger, encoder, clock)
+        ISystemClock clock) : base(options, loggerFactory, encoder, clock)
     {
-        this._options = options;
     }
 
     protected override Task<AuthenticateResult> HandleAuthenticateAsync()
     {
+        this.Logger.LogInformation("Checking API key");
+
         if (string.IsNullOrWhiteSpace(this.Options.ApiKey))
         {
-            return Task.FromResult(AuthenticateResult.Fail("API key not configured on server"));
+            const string ErrorMessage = "API key not configured on server";
+
+            this.Logger.LogError(ErrorMessage);
+
+            return Task.FromResult(AuthenticateResult.Fail(ErrorMessage));
         }
 
         if (!this.Request.Headers.TryGetValue(ApiKeyHeaderName, out StringValues apiKeyFromHeader))
         {
-            return Task.FromResult(AuthenticateResult.Fail("No API key provided"));
+            const string InformationMessage = "No API key provided";
+
+            this.Logger.LogWarning(InformationMessage);
```

Comment:
Why warning here and line 56 and not error?

Reply:
Oh, this one should be Information!  Changed.

The error one is because if you get that log, the backend server is improperly configured.

The warning one is because the server is working properly, but you are being warned someone is trying to get in with a wrong password.